test/SimpleWebRequest.spec.ts

```diff
@@ -178,84 +179,84 @@ describe('SimpleWebRequest', () => {
             const blockSpy = jasmine.createSpy('blockSpy').and.callFake(() => blockDefer.promise());
 
-            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.Critical }, undefined, blockSpy).start().then(onSuccess1);
+            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.Critical }, undefined, blockSpy).start().then(onSuccessCritical);
             jasmine.clock().tick(10);
-            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.High }).start().then(onSuccess2);
+            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.High }).start().then(onSuccessHigh);
             jasmine.clock().tick(10);
 
             SimpleWebRequestOptions.MaxSimultaneousRequests = 1;
             // add a new request to kick the queue
-            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.High }).start().then(onSuccess3);
+            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.High }).start().then(onSuccessHigh2);
             
             expect(blockSpy).toHaveBeenCalled();
 
             jasmine.Ajax.requests.mostRecent().respondWith({ status: 200 });
-            expect(onSuccess2).toHaveBeenCalled();
+            expect(onSuccessHigh).toHaveBeenCalled();
 
-            // unblock the request
+            // unblock the request, it will go back to the queue after the currently executed request
             blockDefer.resolve(void 0);
 
             jasmine.Ajax.requests.mostRecent().respondWith({ status: 200 });
-            expect(onSuccess3).toHaveBeenCalled();
+            expect(onSuccessHigh2).toHaveBeenCalled();
 
             // check if the request at the top of the queue got called again
             expect(blockSpy).toHaveBeenCalledTimes(2);
 
             jasmine.Ajax.requests.mostRecent().respondWith({ status: 200 });
-            expect(onSuccess1).toHaveBeenCalled();
+            expect(onSuccessCritical).toHaveBeenCalled();
         });
 
         it('fails the request, if the blocking promise rejects', done => {
             SimpleWebRequestOptions.MaxSimultaneousRequests = 1;
             const url = faker.internet.url();
             const method = 'GET';
             const blockDefer = SyncTasks.Defer<void>();
-            const onSuccess1 = jasmine.createSpy('onSuccess1');
-            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.Critical }, undefined, () => blockDefer.promise()).start().then(onSuccess1).catch(err => {
-                expect(1).toBe(1);
+            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.Critical }, undefined, () => blockDefer.promise()).start()
+            .then(() => fail(), (err: WebErrorResponse) => {
+                expect(err.statusCode).toBe(0);
+                expect(err.statusText).toBe('Error in _blockRequestUntil: Terrible error');
```

Comment:
I don't think you should have "_blockRequestUntil" in the `statusText`. That property feels too "public API"-like, but a method name feels too "internal API"-like. Consider rewording.

Reply:
How would you name it? It can only happen if the caller has overwritten the `_blockRequestUntil` method, in which case it feels appropriate to point out that this is what's causing the failure